--- install.sh	2024-12-04 02:48:28.696571000 +0000
+++ mod_install.sh	2024-12-04 02:48:28.674645000 +0000
@@ -43,6 +43,7 @@
 esac
 
 VER_PARAM="${OLLAMA_VERSION:+?version=$OLLAMA_VERSION}"
+echo -e "version needed ${VER_PARAM}"
 
 SUDO=
 if [ "$(id -u)" -ne 0 ]; then
@@ -51,7 +52,8 @@
         error "This script requires superuser permissions. Please re-run as root."
     fi
 
-    SUDO="sudo"
+#    SUDO="sudo"
+    SUDO=""
 fi
 
 NEEDS=$(require curl awk grep sed tee xargs)
@@ -63,33 +65,36 @@
     exit 1
 fi
 
-for BINDIR in /usr/local/bin /usr/bin /bin; do
+mkdir -p local/bin
+#for BINDIR in /usr/local/bin /usr/bin /bin; do
+for BINDIR in local/bin; do
     echo $PATH | grep -q $BINDIR && break || continue
 done
-OLLAMA_INSTALL_DIR=$(dirname ${BINDIR})
+OLLAMA_INSTALL_DIR=${BINDIR}
+#OLLAMA_INSTALL_DIR=$(pwd -p)
 
 status "Installing ollama to $OLLAMA_INSTALL_DIR"
-$SUDO install -o0 -g0 -m755 -d $BINDIR
-$SUDO install -o0 -g0 -m755 -d "$OLLAMA_INSTALL_DIR"
+install -m777 -d $BINDIR
+install -m777 -d "$OLLAMA_INSTALL_DIR"
 if curl -I --silent --fail --location "https://ollama.com/download/ollama-linux-${ARCH}.tgz${VER_PARAM}" >/dev/null ; then
     status "Downloading Linux ${ARCH} bundle"
     curl --fail --show-error --location --progress-bar \
         "https://ollama.com/download/ollama-linux-${ARCH}.tgz${VER_PARAM}" | \
-        $SUDO tar -xzf - -C "$OLLAMA_INSTALL_DIR"
+        tar -xzf - -C "$OLLAMA_INSTALL_DIR"
     BUNDLE=1
     if [ "$OLLAMA_INSTALL_DIR/bin/ollama" != "$BINDIR/ollama" ] ; then
         status "Making ollama accessible in the PATH in $BINDIR"
-        $SUDO ln -sf "$OLLAMA_INSTALL_DIR/ollama" "$BINDIR/ollama"
+        ln -sf "$OLLAMA_INSTALL_DIR/ollama" "$BINDIR/ollama"
     fi
 else
     status "Downloading Linux ${ARCH} CLI"
     curl --fail --show-error --location --progress-bar -o "$TEMP_DIR/ollama"\
     "https://ollama.com/download/ollama-linux-${ARCH}${VER_PARAM}"
-    $SUDO install -o0 -g0 -m755 $TEMP_DIR/ollama $OLLAMA_INSTALL_DIR/ollama
+    install -m777 $TEMP_DIR/ollama $OLLAMA_INSTALL_DIR/ollama
     BUNDLE=0
     if [ "$OLLAMA_INSTALL_DIR/ollama" != "$BINDIR/ollama" ] ; then
         status "Making ollama accessible in the PATH in $BINDIR"
-        $SUDO ln -sf "$OLLAMA_INSTALL_DIR/ollama" "$BINDIR/ollama"
+        ln -sf "$OLLAMA_INSTALL_DIR/ollama" "$BINDIR/ollama"
     fi
 fi
 
@@ -105,7 +110,7 @@
 configure_systemd() {
     if ! id ollama >/dev/null 2>&1; then
         status "Creating ollama user..."
-        $SUDO useradd -r -s /bin/false -U -m -d /usr/share/ollama ollama
+        $SUDO useradd -r -s /bin/false -U -m -d share/ollama ollama
     fi
     if getent group render >/dev/null 2>&1; then
         status "Adding ollama user to render group..."
